crates/vm/levm/src/opcode_handlers/block.rs

```diff
@@ -371,11 +371,14 @@ impl VM {
         &mut self,
         current_call_frame: &mut CallFrame,
     ) -> Result<OpcodeSuccess, VMError> {
-        if self.env.consumed_gas + gas_cost::GASPRICE > self.env.gas_limit {
+        if self.env.consumed_gas + gas_cost::GASPRICE > self.env.tx_env.gas_limit {
             return Err(VMError::OutOfGas);
         }
 
-        current_call_frame.stack.push(self.env.gas_price)?;
+        // TODO: if not legacy or access list, then gas price is max_fee_per_gas
```

Comment:
I can create it, but with the PR that Luca is working on right now it will automatically be fixed, as the change he is implementing will have gas price to be created from `gas_price` or `max_fee_per_gas ` depending on the transaction type. What do you think I should do?

Reply:
We can remove the TODO from this PR, or make sure that this TODO is removed in said PR. Btw, can you reference the PR you're talking about?